Tools/Driver.PackOne.targets

```diff
@@ -0,0 +1,203 @@
+        foreach (var item in Item)
+        {
+            string curr = item.GetMetadata("Identity");
+            if (Value==curr)
+            {
+                Result = true;
+                break;
+            }
+        }
+        return true;
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <!-- checking $(TargetOS) and $(TargetArch) -->
+  <Target Condition="'$(Feature_PackOne)'=='true'" Name="CheckPackOne" BeforeTargets="PackOneTarget">
+    <ItemGroup>
+      <PackOne_ValidOS Include="Wxp;Wnet;Wlh;Win7;Win8;Win8.1;Win10"/>
+      <PackOne_ValidArch Include="x86;amd64"/>
+    </ItemGroup>
+    <Message Text="Checking PackOne: TargetOS=$(TargetOS) and TargetArch=$(TargetArch)" Importance="high"/>
+    <Contained Value="$(TargetOS)" Item="@(PackOne_ValidOS)">
+      <Output TaskParameter="Result" PropertyName="PackOne_IsValidOS"/>
+    </Contained>
+    <Contained Value="$(TargetArch)" Item="@(PackOne_ValidArch)">
+      <Output TaskParameter="Result" PropertyName="PackOne_IsValidArch"/>
+    </Contained>
+    <Contained Value="$(TargetOS)" Item="@(PackOne_UnsupportedOS)">
+      <Output TaskParameter="Result" PropertyName="PackOne_IsUnsupportedOS"/>
+    </Contained>
+    <Message Text="TODO CheckPackOne: found IsValidOS=$(PackOne_IsValidOS) and IsValidArch=$(PackOne_IsValidArch) and IsUnsupportedOS=$(PackOne_IsUnsupportedOS)" Importance="high"/>
+    <Error Text="PackOne: Wrong OS parameter $(TargetOS)" Condition="'$(PackOne_IsValidOS)'=='false'" />
+    <Error Text="PackOne: Wrong Arch parameter $(TargetArch)" Condition="'$(PackOne_IsValidArch)'=='false'" />
+    <Error Text="PackOne: Unsupported OS parameter $(TargetOS)" Condition="'$(PackOne_IsUnsupportedOS)'=='true'" />
+  </Target>
+
+  <!-- default driver name -->
+  <PropertyGroup Condition="'$(PackOne_FileName)'==''">
+    <PackOne_FileName>$(TargetName)</PackOne_FileName>
+  </PropertyGroup>
+
+  <!-- Win8.1 $(TargetOS) is unsupported by PackOne -->
+  <ItemGroup>
+    <PackOne_UnsupportedOS Include="@(PackOne_UnsupportedOS);Win8.1" />
+  </ItemGroup>
+
+  <!-- default mask customization options -->
+  <PropertyGroup Condition="'$(PackOne_WlhAddWin7Mask)'==''">
+    <PackOne_WlhAddWin7Mask>true</PackOne_WlhAddWin7Mask>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(PackOne_Win7AddWlhMask)'==''">
+    <PackOne_Win7AddWlhMask>false</PackOne_Win7AddWlhMask>
+  </PropertyGroup>
+
+  <!-- from packone.bat create_xp variant
+       both Wxp and Wnet drivers have mask for both Windows XP and Server2003 OSes -->
+  <PropertyGroup Condition="'$(TargetOS)'=='Wxp' AND '$(TargetArch)'=='x86'">
+    <PackOne_OsMask>XP_X86,Server2003_X86</PackOne_OsMask>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetOS)'=='Wxp' AND '$(TargetArch)'=='amd64'">
+    <PackOne_OsMask>XP_X64,Server2003_X64</PackOne_OsMask>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetOS)'=='Wnet' AND '$(TargetArch)'=='x86'">
+    <PackOne_OsMask>XP_X86,Server2003_X86</PackOne_OsMask>
```

Comment:
This is how packone.bat is currently written.
Let change current behaviour later?